fixed file separator for windows, diamond notation

programming2web opened the pull request:
Fixed file separation symbol to run cross-platform
Fixed diamond notation and generics
Resolved unchecked warnings - all except one in FileTreeModel which I had to suppress (the cast is safe)
Removed “FileSieve” label
Added minimum width for main frame

Please let me know if it's Ok to merge. Will work on the rest on Wednesday - Thursday. Thanks.

programming2web committed:
fixed file separator for windows, diamond notation

Fixed file separation symbol to run cross-platform
Fixed diamond notation
Resolved unchecked warnings - all except one in FileTreeModel which I
suppressed as the cast is safe
Removed “FileSieve” label
Added minimum width for main frame

smarks commented:
Hi,

It looks like my pull request (which just contained comments) is not
around. Has anyone seen it?

Spencer

On Mon, Jul 21, 2014 at 10:20 PM, Olga notifications@github.com wrote:

> Fixed file separation symbol to run cross-platform
> Fixed diamond notation and generics
> Resolved unchecked warnings - all except one in FileTreeModel which I had
> to suppress (the cast is safe)
> Removed “FileSieve” label
> Added minimum width for main frame
> 
> Please let me know if it's Ok to merge. Will work on the rest on Wednesday
> 
> ## \- Thursday. Thanks.
> 
> You can merge this Pull Request by running
> 
>   git pull https://github.com/JeffBlodgett/FileSieve GUI-and-Persistence
> 
> Or view, comment on, or merge it at:
> 
>   https://github.com/JeffBlodgett/FileSieve/pull/19
> Commit Summary
> - fixed file separator for windows, diamond notation
> 
> File Changes
> - _M_ src/main/java/FileSieve/gui/Controller.java
>   https://github.com/JeffBlodgett/FileSieve/pull/19/files#diff-0 (18)
> - _M_ src/main/java/FileSieve/gui/CopyScreen.java
>   https://github.com/JeffBlodgett/FileSieve/pull/19/files#diff-1 (8)
> - _M_ src/main/java/FileSieve/gui/FileTreeModel.java
>   https://github.com/JeffBlodgett/FileSieve/pull/19/files#diff-2 (23)
> - _M_ src/main/java/FileSieve/gui/ScreenSwitcher.java
>   https://github.com/JeffBlodgett/FileSieve/pull/19/files#diff-3 (25)
> - _M_ src/test/java/FileSieve/gui/CheckTreeManagerTest.java
>   https://github.com/JeffBlodgett/FileSieve/pull/19/files#diff-4 (5)
> - _M_ src/test/java/FileSieve/gui/ControllerTest.java
>   https://github.com/JeffBlodgett/FileSieve/pull/19/files#diff-5 (22)
> - _M_ src/test/java/FileSieve/gui/FileTreeModelTest.java
>   https://github.com/JeffBlodgett/FileSieve/pull/19/files#diff-6 (8)
> - _T_ src/test/java/FileSieve/gui/util/UtilitiesTest.java
>   https://github.com/JeffBlodgett/FileSieve/pull/19/files#diff-7 (0)
> 
> Patch Links:
> - https://github.com/JeffBlodgett/FileSieve/pull/19.patch
> - https://github.com/JeffBlodgett/FileSieve/pull/19.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/JeffBlodgett/FileSieve/pull/19.

## 

Spencer Marks
http://smarksphotography.com

JeffBlodgett commented on src/main/java/FileSieve/gui/Controller.java:
```diff
@@ -414,14 +414,14 @@ protected void saveDiffReport(){
                 displayAlert("Couldn't create the report");
             }
             try {
-                diffReport.save(targetPath+"/FileSieveDiffReport.html");
```
This line assumes the user was aware that he/she was only supposed to have used the file chooser to select an existing folder within which to place the diff report (whose name is hardcoded here).  Per Hangouts discussion, I have posted sample code in out Blackboard discussion area for a JFileChooser that, "enforces the selection/specification of a file name with the enforcement of one of a number of specific, predefined 3-character extensions."

programming2web committed:
GUI Fine Tuning

- Replaced file chooser to select destination for saving reports
- Included deleted files info in report
- Added scrolling to the bottom of the list on copy screen as files are
being copied
- Replaced text “copying …” to “creating …” per Jeff’s suggestion
- Added instructions about selecting multiple folders on select screen

programming2web commented:
Thank you, Jeff. I left you a reply on Blackboard discussion. Really appreciate your help with this and all your valuable input.

The last push covers the following:
- Replaced file chooser to select destination for saving reports
- Included deleted files info into report
- Added scrolling to the bottom of the list on copy screen as files are being copied
- Replaced text “copying …” to “creating …” on copy screen per Jeff’s suggestion
- Added instructions on how to select multiple folders on select screen

Let me know if you think it is ready for merging or still needs some tweaking. Thanks!

I'll probably have an hour or two tomorrow to do the final touches and maybe an hour on Sunday. Sorry that I can't dedicate more time to it on its final stage... I do try my best.

programming2web commented:
Forgot to mention - saving info about deleted files works like this:
1. If user clicks "Export report" before deleting any files - no info about deleted files is included in the report (maybe he doesn't intend to delete them and just wants to see what duplicates he has or maybe he wants only to copy them)
2. If user deletes some files and then exports report - info about deleted files will be included in the report
3. If user then deletes some more files - all of the deleted files will be included in the report
4. If user starts a new duplicates search - all info about previous duplicate files results (including the files that were deleted and saved or not saved in report) is gone. Only newly found duplicates will be included in the report. So, each report corresponds to one duplicate search.

Hope it makes sense.

dbendig commented:
Thanks, Olga. Everything looks great. The only thing I noticed is that the default file extension for the report is still html, but I believe from our chats earlier it was said we should be using the three character htm extension. If I'm wrong about that then this should be merged.

JeffBlodgett commented:
That works for me too Olga!  Looks like you modified a saveFileChooser to only support selection/specification of a file and also set the default (suggested) file name.  If we had more time I’d suggest enforcing the htm (or html) file extensions as in sample code I posted to Blackboard but what you have works nicely.  I had to change the following line to get things to compile but suspect that is because you are working atop a newer persistence/reports package:

```
diffReport = DiffReportFactory.getDiffReport(duplicates); //, deletedPaths);
```

I suggest we all merge what we have thus far (persistence and reports, GUI, and my pull request from yesterday).  Give me the green light and I’ll merge my pull.  You both have mine.  I’m working on BusinessLogic again today (documentation and seeing if I can’t use a Proxy(s) to make my DiscoveredPath instances appear more like the system-generated Path instances they decorate… almost sure I can’t do this but looking into it anyway as I was Googling from bed last evening and found something interesting).
- Jeff

From: dbendig [mailto:notifications@github.com] 
Sent: Thursday, July 24, 2014 12:33 AM
To: JeffBlodgett/FileSieve
Cc: Jeff Blodgett
Subject: Re: [FileSieve] fixed file separator for windows, diamond notation (#19)

Thanks, Olga. Everything looks great. The only thing I noticed is that the default file extension for the report is still html, but I believe from our chats earlier it was said we should be using the three character htm extension. If I'm wrong about that then this should be merged.

—
Reply to this email directly or view it on GitHub https://github.com/JeffBlodgett/FileSieve/pull/19#issuecomment-49966563 .

dbendig commented:
If Jeff doesn't see the changes to the Reports then this branch should be merged with Report-Update before it is merged into master.

programming2web commented:
Thanks, guys. I'm very happy it work on Windows now. I'll comment out the call to overloaded DiffReportFactory.getDiffReport(duplicates, deletedPaths) in order not to break the build and will merge shortly. Dan, once you merge your changes to master I'll put this call back. 
Dan, the reason the extension is html and not htm is because unfortunately I didn't use all the code that Jeff has suggested as it would take me a lot of time to write proper tests for it. If used without the tests I'm afraid it would decrease our project test coverage considerably. So instead I used a much simpler version which doesn't validate file extension, so any extension for the report will work.

dbendig commented:
Olga, could you commit the changes to the Report-Update branch then? I don't think there should be multiple pull requests for the same feature.

programming2web commented:
Ok. Merging to Report-Update.

JeffBlodgett commented:
Ok - think I understand now... the work associated with Dan (current) pull request has been merged into this branch (right?).  I'll pull this branch now and run atop Windows to ensure all is running well... ok, no it doesn't seem as though Dan's changes have been merged into this branch yet.  Should he merge his pull request and then you could merge this one Olga?  This would enable the project's master to generate Dan's current DiffReport, incorporate your changes/fixes as noted, and permit me to test to ensure my updates to the business logic this week (which are already integrated into the master branch) are not causing any problem atop my Windows platform.  This won't preclude you from doing any additional work this weekend.  I am filling out some more documentation-relating items today myself.

JeffBlodgett commented:
This is what the compilers shows when I try to compile this current GUI-and-Persistence branch, as applied atop the current master... this is what makes me think the changes in Dan's pull request have not been integrated:

compile:
    [javac] Compiling 47 source files to C:\Users\Jeffrey\SkyDrive\IDEA Projects\FileSieve\build\classes
    [javac] C:\Users\Jeffrey\SkyDrive\IDEA Projects\FileSieve\src\main\java\FileSieve\gui\Controller.java:440: error: method getDi
ffReport in class DiffReportFactory cannot be applied to given types;
    [javac]                     diffReport = DiffReportFactory.getDiffReport(duplicates, deletedPaths);
    [javac]                                                   ^
    [javac]   required: List<SimpleImmutableEntry<String,List<File>>>
    [javac]   found: List<SimpleImmutableEntry<String,List<File>>>,List<String>
    [javac]   reason: actual and formal argument lists differ in length
    [javac] 1 error

programming2web commented:
Jeff, all my changes are in Dan's pull request now for his branch "Report-Update", pull request #22.
When he merges his branch all my and his changes will be in the master. I will just close this pull request in order not to confuse anyone since I won't merge it now.

JeffBlodgett commented:
Ok - thanks Olga!  By the way, thanks for creating the Jar and Run targets in build.xml!
